server/src/main/java/io/crate/expression/operator/EqOperator.java

```diff
@@ -157,7 +157,7 @@ public static Query termsQuery(String column, DataType<?> type, Collection<?> va
         }
         List<?> nonNullValues = values.stream().filter(Objects::nonNull).toList();
         if (nonNullValues.isEmpty()) {
-            return null;
+            return NumTermsPerDocQuery.forColumn(column, type, numTerms -> numTerms == 0);
```

Comment:
Does not work:
```
cr> create table t(a int[][]);
CREATE OK, 1 row affected (1.585 sec)

cr> insert into t values ([[], [1,2,3]]);
INSERT OK, 1 row affected (0.362 sec)

cr> select * from t where a[1] = [];
+---+
| a |
+---+
+---+
SELECT 0 rows in set (2.208 sec)
```
The row is filtered because 1, 2, 3 are counted.